src/Tickets/Commerce/Gateways/Free/Gateway.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Tickets Commerce: Free Gateway.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Free
+ */
+
+namespace TEC\Tickets\Commerce\Gateways\Free;
+
+use TEC\Tickets\Commerce\Cart;
+use TEC\Tickets\Commerce\Gateways\Contracts\Abstract_Gateway;
+use TEC\Tickets\Commerce\Module;
+use Tribe__Template as Template;
+
+/**
+ * Class Free Gateway.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Gateways\Free
+ */
+class Gateway extends Abstract_Gateway {
+	/**
+	 * The Gateway key.
+	 *
+	 * @since TBD
+	 *
+	 * @var string $key The Gateway key.
+	 */
+	protected static $key = 'free';
+	
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_label() {
+		return '';
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public static function is_enabled(): bool {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function is_connected() {
+		return true;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function is_active() {
+		return tribe( Module::class )->is_checkout_page() && static::should_show();
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function should_show() {
+		$cart_total = tribe( Cart::class )->get_cart_total();
+		return 0 == $cart_total;
+	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function get_admin_notices() {
+		return [];
+	}
+	
+	/**
+	 * Render the checkout template.
+	 *
+	 * @since TBD
+	 *
+	 * @param Template $template The template object.
+	 */
+	public function render_checkout_template( Template $template ): string {
```

Comment:
```suggestion
	public function render( Template $template ): string {
```
My preference would be this function to move to a `View` or `Template` with just a `render()`. Seems like this is responsible for too many things.

Again, perhaps beyond scope at this point but want to mention it.

Reply:
Yes, out of scope as this is coming from the [abstract](https://github.com/the-events-calendar/event-tickets/blob/a40a7e3d262771d5fa54438c732936d8e47b84e9/src/Tickets/Commerce/Gateways/Contracts/Abstract_Gateway.php#L315).